lucet-runtime/lucet-runtime-internals/src/instance.rs

```diff
@@ -530,9 +525,43 @@ impl Instance {
         self.swap_and_return()
     }
 
+    /// Run the module's [start function][start], if one exists.
+    ///
+    /// If there is no start function in the module, this does nothing.
+    ///
+    /// If the module contains a start function, you must run it before running any other exported
+    /// functions. If an instance is reset, you must run the start function again.
+    ///
+    /// Start functions may assume that Wasm tables and memories are properly initialized, but may
+    /// not assume that imported functions or globals are available.
+    ///
+    /// # Errors
+    ///
+    /// In addition to the errors that can be returned from [`Instance::run()`][run], this can also
+    /// return `Error::StartYielded` if the start function attempts to yield. This should not arise
+    /// as long as the start function does not attempt to use any imported functions.
+    ///
+    /// # Safety
+    ///
+    /// The foreign code safety caveat of [`Instance::run()`][run]
+    /// applies.
+    ///
+    /// [run]: struct.Instance.html#method.run
+    /// [start]: https://webassembly.github.io/spec/core/syntax/modules.html#syntax-start
+    pub fn run_start(&mut self) -> Result<(), Error> {
```

Comment:
should `run_start` return an error if it has already been run and the instance not reset? I don't think it's terribly likely to have been calling the start function manually before, so accidentally rerunning it was unlikely, but double-initialization seems more likely with this as public API.

Reply:
Yeah, I actually was just thinking this. Good call

Reply:
Not a blocking concern, just a question that I'm curious about: would it be feasible to return an error if any exports are called on an instance before its start function has run?

Reply:
Done in 21a8ff3 :sparkles:

Reply:
> Not a blocking concern, just a question that I'm curious about: would it be feasible to return an error if any exports are called on an instance before its start function has run?

Aaa, race condition on the comments, sorry. This is a good idea; I will add it while my brain is in this corner of the project.

Reply:
@sunfishcode your suggestion is implemented in 50f808f. Thank you!

Reply:
If I read the patch right, it's issuing an error if the start function calls any imports. My question was about issuing errors if any exported function in a module is called before the module's start function is called, which isn't normally possible.

Reply:
Well, I definitely read that wrong, whoops. Yes, an error is returned if you try to call any other entrypoints before running the start function, if it exists.

Should I keep the terminate-on-import-function behavior? Now that I'm looking at the spec again I can't actually figure out whether import functions can be called in the start section.

Reply:
Imports can be called from the start function.

Reply:
Where's my dunce cap emoji? Reverted in 1e84d31